docs/polkadot/implementors-guide/guide.md

```diff
@@ -63,7 +63,22 @@ Note that the candidate can fail to be included in any of the following ways:
+This brings us to the second part of the process. Once a parablock is "fully included", it is still "pending approval". At this stage in the pipeline, the parablock has been backed by a majority of validators in the group assigned to that parachain, and its data has been guaranteed available by the set of validators as a whole. However, the validators in the parachain-group (known as the "Parachain Validators" for that parachain) are sampled from a validator set which contains some proportion of byzantine, or arbitrarily malicious members. This implies that the Parachain Validators for some parachain may be majority-dishonest, which means that secondary checks must be done on the block before it can be considered approved.
+
+The Approval Process looks like this:
+1. Parablocks are pending approval for a time-window known as the secondary checking window.
+1. During the secondary-checking window, validators randomly self-select to perform secondary checks on the parablock.
+1. These validators, known in this context as secondary checkers, acquire the parablock and its PoV, and re-run the validation function.
+1. The secondary checkers submit the result of their checks to the relay chain. Contradictory results lead to escalation, where even more secondary checkers are selected and the secondary-checking window is extended.
+1. At the end of the Approval Process, the parablock is either Approved or it is rejected. More on the rejection process later.
+
+These two pipelines sum up the sequence of events necessary to extend and acquire full security on a Parablock. Note that the Inclusion Pipeline must conclude for a specific parachain before a new block can be accepted on that parachain. After inclusion, the Approval Process kicks off, and can be running for many parachain blocks at once.
+
+[TODO Diagram: Inclusion Pipeline & Approval Processes interaction]
+
+It is also important to take note of the fact that the relay-chain is extended by BABE, which is a forkful algorithm. That means that different block authors can be chosen at the same time, and may not be building on the same block parent. Furthermore, the set of validators is not fixed, nor is the set of parachains. And even with the same set of validators and parachains, the validators' assignments to parachains is flexible. This means that the architecture proposed in the next chapters must deal with the variability and multiplicity of the network state.
```

Comment:
It might go without saying that while the code does need to deal with all this, an important design goal is to minimize disagreement between forks on things like parachain assignments.